### PR TITLE
Make sure to test Unicode handling

### PR DESCRIPTION
**Pull request by twaugh**
Signed-off-by: Tim Waugh <twaugh@redhat.com>

### CODE REVIEW

**Comment by lcarva** [comment]
Is this still relevant?

**Comment by twaugh** [comment]
I think it is, just need some time to look at it.

File: tests/test_parser.py [hunk 1]
```diff
@@ -46,21 +46,31 @@ def read_version(fp, regex):
...
         assert dfparser.lines == df_lines
+        assert [isinstance(line, six.text_type) for line in dfparser.lines]
 
         dfparser.content = ""
         dfparser.lines = df_lines
         assert dfparser.content == df_content
         assert dfparser.lines == df_lines
+        assert [isinstance(line, six.text_type) for line in dfparser.lines]
+
+        filename = os.path.join(str(tmpdir), 'Dockerfile')
```
Comment (lcarva) [inline tests/test_parser.py]: Something I discover the other day:
```python
tmpdir.join('Dockerfile').write_text(df_content, 'utf-8')
```

I haven't actually tried using `write_text`, but I think joining and writing like that is pretty :cool:
http://py.readthedocs.io/en/latest/path.html#py._path.local.LocalPath.write_text

Comment (twaugh) [reply tests/test_parser.py]: Yes, that's nice. Updated.

**Comment by lcarva** [comment]
Not updating release notes draft because these are unit test changes.

**Pull request merged**